core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -915,43 +915,50 @@ private static void matchJoin(SubQueryRemoveRule rule, RelOptRuleCall call) {
     int nFieldsLeft = join.getLeft().getRowType().getFieldCount();
     int nFieldsRight = join.getRight().getRowType().getFieldCount();
 
-    if (inputSet.intersects(ImmutableBitSet.range(0, nFieldsLeft))
-        && inputSet.intersects(ImmutableBitSet.range(nFieldsLeft, nFieldsLeft + nFieldsRight))) {
+
+    ImmutableBitSet inputIntersectsLeftSide = ImmutableBitSet.range(0, nFieldsLeft);
```

Comment:
What is meant is to save the boolean value directly in a final variable, if you prefer to keep this form let's at least adapt the name of the variables to `leftInputFields`/`rightInputFields` and make them final.